libavfilter/vf_drawbox.c

```diff
@@ -186,6 +202,8 @@ static int config_input(AVFilterLink *inlink)
     int ret;
     int i;
 
+    if (s->box_source_expr && (ret = box_source_expr_parse(expr = s->box_source_expr, &s->box_source)))
```

Comment:
looks that expression is not good here, and also in the 'detection through tensorflow' patch.  As an expression, it should accept variables such as iw, ih, ....

Maybe just s->box_source here.